From d2811f8168df731a4bdb241ef8e297c89826039e Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Thu, 08 Oct 2020 08:35:17 +0300
Subject: [PATCH] [SW-20979] goya2: tstd: fix rounding issues

using s32 instead of u32 to avoid rounding issues due to negative values

Change-Id: I6d15867a73abd8aa0223d0cf852ae993c318a427
Signed-off-by: Sailaja <sbandarupalli@habana.ai>
---

diff --git a/preboot/thermal_v2.c b/preboot/thermal_v2.c
index 62aa3ce..7e85c82 100644
--- a/preboot/thermal_v2.c
+++ b/preboot/thermal_v2.c
@@ -212,7 +212,7 @@
 long hl_tmp_temp_reg_to_mC(int val, int ts)
 {
 	long tmp = (val & HL_TS_00_SDIF_DATA_SMPL_DATA_MASK);
-	u32 coef_a_g, coef_b_h, coef_div;
+	s32 coef_a_g, coef_b_h, coef_div;
 
 	coef_a_g = hl_ts_data[ts].hl_tmp_data.coef_a_g;
 	coef_b_h = hl_ts_data[ts].hl_tmp_data.coef_b_h;
@@ -235,7 +235,7 @@
 u32 hl_tmp_mC_to_temp_reg(long val, int ts)
 {
 	u32 reg_val;
-	u32 coef_a_g, coef_b_h, coef_div;
+	s32 coef_a_g, coef_b_h, coef_div;
 
 	coef_a_g = hl_ts_data[ts].hl_tmp_data.coef_a_g;
 	coef_b_h = hl_ts_data[ts].hl_tmp_data.coef_b_h;
